pkg/controller/backup_controller.go

```diff
@@ -865,3 +878,42 @@ func encodeToJSONGzip(data interface{}, desc string) (*bytes.Buffer, []error) {
 
 	return buf, nil
 }
+
+// Waiting for VolumeSnapshot ReadyTosue to true is time consuming. Try to make the process parallel by
+// using goroutine here instead of waiting in CSI plugin, because it's not easy to make BackupItemAction
+// parallel by now. After BackupItemAction parallel is implemented, this logic should be moved to CSI plugin
+// as https://github.com/vmware-tanzu/velero-plugin-for-csi/pull/100
+func (c *backupController) checkVolumeSnapshotReadyToUse(volumesnapshots []*snapshotv1api.VolumeSnapshot) error {
```

Comment:
IMO  it's better set the context as the first param of this func so when we refactor controller the context of reconcile can be passed.

Reply:
Modified.